nautobot/utilities/testing/__init__.py

```diff
@@ -35,12 +44,43 @@
 )
 
 
+def run_job_for_testing(job, data=None, commit=True, username="test-user"):
```

Comment:
Would you be willing to look at `nautobot/extras/tests/test_jobs.py` and see if the existing code there (particularly `create_job_result_and_run_job`) could be adjusted to make use of this new utility function? Not blocking for this PR but it'd be nice to have.

Reply:
I gave this a quick pass, judging from the passing tests it looks good!